### ISSUE TITLE
Issues in translatable messages in R code

### ISSUE THREAD

**Issue by aitap** [Rdatatable/data.table#6532]
### Fragmented sentences

* [x] `%s` is `elements of fun.list` or ...`arguments to measure` or maybe `measured columns`, untranslated, coming from the `group.desc` argument, which is a documented ("for internal use") interface. Another `%s` is `number of capture groups in pattern` or `max number of items after splitting column names`, also untranslated.
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/fmelt.R#L91
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/fmelt.R#L97
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/fmelt.R#L102
* [x] `%1$s` is `keys` or `indices`, untranslated.
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/setops.R#L160-L177
* [x] `%3$s` is either `error` or `warning`, untranslated. That's relatively minor and could be worked around in translation (`condition of class %s`):
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/fread.R#L314

### Minor issues with correctness

* [x] This should have said `POSIXct`, not just `POSIX` (which could be read as `POSIXt` or `POSIXlt`, and we don't like `POSIXlt`):
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/between.R#L13
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/between.R#L15
* [x] Missing `]` at the end:
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/data.table.R#L295
* [x] The last `}` needs to be replaced with `]`
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/data.table.R#L412
* [x] [PR17478](https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17478) is now `FIXED CLOSED`:
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/onLoad.R#L27

### Nitpicking

* [x] In R ≥ 4.2.0 there's the [`Sys.setLanguage`](https://search.r-project.org/R/refmans/base/html/gettext.html) function which (tries to) set the `gettext` language during runtime, maybe recommend it here?
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/onAttach.R#L28
* [x] Looks like the code accepts a _single_ `data.table`/`data.frame`, maybe remove the `s`?
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/duplicated.R#L105
* [x] `s/both libraries/both packages/`
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/fmelt.R#L17
* [x] Maybe shouldn't be translated
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/setops.R#L9
* [x] Needs `# nontranslate` because it's from a different domain
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/test.data.table.R#L118
* [x] Perhaps ...`vector **of** length %d`?
https://github.com/Rdatatable/data.table/blob/745160d7247170b7910324ae7c2f25c8d5cc4cf6/R/data.table.R#L1043

**Comment by MichaelChirico** [comment]
@tdhock can you help me understand `measurev()` here? It has an argument `group.desc=` with a default value `"elements of fun.list"`. But I only ever see `measurev()` called in one place, with a different value:

https://github.com/Rdatatable/data.table/blob/6abbaafdb4839810b8e19f762f227b594329cfd9/R/fmelt.R#L70

Can we just drop the `group.desc` argument, since `measurev()` is unexported? Also, remind me why it is documented given that it's unexported?

**Comment by tdhock** [comment]
since measure is exported, we should export measurev too. both are documented because they are meant to be used as the `measure.vars` argument to `melt`.
when I wrote measure/measurev, I was not aware of the issues created by breaking strings apart when translating messages.
so we should probably get rid of the `group.desc` argument, I will look at #6534

**Comment by MichaelChirico** [comment]
OK, filed #6535.

**Comment by MichaelChirico** [comment]
I see now in the tests that `measurev()` is called with the default `group.desc=`, I guess some NSE applies there.

**Comment by tdhock** [comment]
the user could call either measure() or measurev() inside `measure.vars`.
`measure()` calls `measurev()` but we want specific error messages to help the user:
```r
> melt(data.table(iris)[1], measure.vars=measure(part, part))
Error in err.names.unique(group.desc, names(fun.list)) : 
  ... arguments to measure should be uniquely named, problems: [part]
> melt(data.table(iris)[1], measure.vars=measurev(list(part=NULL, part=NULL)))
Error in err.names.unique(group.desc, names(fun.list)) : 
  elements of fun.list should be uniquely named, problems: [part]
```

**Comment by MichaelChirico** [comment]
Yea, I agree it's not _perfect_, but it certainly looks actionable to me because `part` is clearly duplicated.

**Comment by rikivillalba** [comment]
In deve.R we can merge
https://github.com/Rdatatable/data.table/blob/11a44bd5a31271ec176ff5b8134458ec2f153917/R/devel.R#L40:L44
into
```
    pkg_rev <- unname(read.dcf(system.file("DESCRIPTION", package=pkg, lib.loc=lib, mustWork=TRUE), fields=field)[, field])
	if(upg) {
      catf("R %s package %s has been updated to (%s)\n",  pkg, pkg_rev, utils::packageVersion(pkg, lib.loc=lib))
    } else {
      catf("R %s package %s is up-to-date at (%s)\n",  pkg, pkg_rev, utils::packageVersion(pkg, lib.loc=lib))
    }
```
I think we can use `catf` despite the previous line unloads the package.

**Comment by aitap** [comment]
This is tricky code. Objects that live in package namespaces are "lazy-loaded": the namespace environment initially contains "promise" objects that, when first accessed, quickly seek to pre-remembered offsets in `R/data.table.rdb` and uncompress whatever they find there. This function unloads the namespace and installs a new package, replacing the `*.rdb` file with a different one, where the offsets have changed. If the `catf` function is still a promise at this point, it will at least cause a warning (`internal error -3 in R_decompress1`), maybe fail to evaluate (`lazy-load DB is corrupt`):

```
Downloaded package sources are in
        ‘/tmp/Rtmp3GzX83/downloaded_packages’
R data.table package 0c957c70d019ebf36c44c988d5b2a4def9c05c92 has been updated to (1.16.99)
Warning:
In get(method, envir = home) : internal error -3 in R_decompress1
```

It may be better to limit the "last on the way out" code to base-R functions only:

```r
    pkg_rev = unname(read.dcf(system.file("DESCRIPTION", package=pkg, lib.loc=lib, mustWork=TRUE), fields=field)[, field])
    cat(sprintf(
      if (upg) gettext(
        "R %s package %s has been updated to (%s)\n", domain = "R-data.table"
      ) else gettext(
        "R %s package %s is up-to-date at (%s)\n", domain = "R-data.table"
      ),
      pkg, pkg_rev, utils::packageVersion(pkg, lib.loc=lib)
    ))
```

**Comment by MichaelChirico** [comment]
Closed between #6534 and #6536
